subsys/bluetooth/host/conn.c

```diff
@@ -1204,14 +1204,36 @@ void bt_conn_recv(struct bt_conn *conn, struct net_buf *buf, uint8_t flags)
 	/* Check packet boundary flags */
 	switch (flags) {
 	case BT_ACL_START:
+		if (conn->rx) {
+			BT_ERR("Unexpected first L2CAP frame");
+			bt_conn_reset_rx_state(conn);
+		}
+
+		if (!buf->len) {
```

Comment:
I'm a bit unsure whether it is allowed or not for the controller to send zero-length ACL data to the host. The other way is definitely allowed (and is tested in LL conformance tests). In any case, the host shouldn't misbehave if it does receive zero-length ACL data.

Reply:
I guess 0 length is also possible, at least I don't see anything on the spec preventing that.